app/src/lib/databases/repositories-database.ts

```diff
@@ -21,6 +21,8 @@ export interface IDatabaseGitHubRepository {
 
   /** The last time a prune was attempted on the repository */
   readonly lastPruneDate: number | null
+
+  readonly permissions: 'read' | 'write' | 'admin' | null
```

Comment:
I _think_ this needs to be `permissions?` since we're not doing any database migration to ensure that it's present on the existing database entries. If we don't I think we could end up having `GitHubRepository` instances where the `permissions` property is undefined even though the type says that it can only be one of the three strings or `null`.

```suggestion
  readonly permissions?: 'read' | 'write' | 'admin' | null
```

Reply:
you are correct! thanks for catching that. updated in https://github.com/desktop/desktop/pull/8635/commits/09e35e9ea63ef2cb62bf6a89d1c10217c5ea17ee